File: Chess/src/main/java/game/board/Tile.java
package game.board;

import game.piece.Coordinate;
import game.piece.Piece;

public class Tile implements java.io.Serializable {
    public Piece piece;
    public Coordinate coordinate;

    public Tile(Coordinate coordinate, Piece piece) {
        this.setPiece(piece);
        this.setCoordinate(coordinate);
    }

    public Piece getPiece() {
        return this.piece;
    }

    public void setPiece(Piece p) {
        this.piece = p;
    }

    public Coordinate getCoordinate() {
        return this.coordinate;
    }

    public void setCoordinate(Coordinate coordinate) {
        this.coordinate = coordinate;
    }

    public boolean hasPiece() {
        return this.piece != null;
    }

    @Override
    public String toString() {
        return coordinate.toString() + " Piece " + ((hasPiece() ? piece.toString() : "Empty"));
    }

}

File: Chess/src/main/java/game/board/Tile.kt
package game.board

import game.piece.Coordinate
import game.piece.Piece
import game.piece.PieceNull
import java.io.Serializable

open class Tile(var coordinate: Coordinate, var piece: Piece) : Serializable {
    fun hasPiece(): Boolean {
        return piece !is PieceNull
    }

    override fun toString(): String {
        return coordinate.toString() + " Piece " + if (hasPiece()) piece.toString() else "Empty"
    }
}